core/services/workflows/engine.go

```diff
@@ -195,6 +206,55 @@ func (e *Engine) init(ctx context.Context) {
 	e.afterInit(true)
 }
 
+var (
+	defaultOffset, defaultLimit = 0, 1_000
+)
+
+func (e *Engine) resumeInProgressExecutions(ctx context.Context) error {
+	wipExecutions, err := e.executionStates.GetUnfinished(ctx, defaultOffset, defaultLimit)
+	if err != nil {
+		return err
+	}
+
+	// TODO: paginate properly
+	if len(wipExecutions) == defaultLimit {
+		e.logger.Warnf("possible execution overflow during resumption")
+	}
```

Comment:
```suggestion
	if len(wipExecutions) >= defaultLimit {
		e.logger.Warnf("possible execution overflow during resumption")
	}
```

Is it possible for pending executions between function calls to skip above the defaultLimit?

Reply:
I don't think so (I'd expect a LIMIT to always return maximum that amount of records), but I can change it to be safe.